src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/RegionDropdown.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {FC, useCallback, useContext, useMemo} from 'react'
+import {ComponentStatus, Dropdown} from '@influxdata/clockface'
+
+// Components
+import {CreateOrgContext} from 'src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationContext'
+import {Cluster} from 'src/client/unityRoutes'
+
+const SINGLE_PROVIDER_WIDTH = 375
+
+export const RegionDropdown: FC = () => {
+  const {clusters, currentProvider, currentRegion, changeCurrentRegion} =
+    useContext(CreateOrgContext)
+  const handleRegionChange = useCallback(
+    region => {
+      changeCurrentRegion(region)
+    },
+    [changeCurrentRegion]
+  )
+  const regions = useMemo(() => {
+    return clusters?.[currentProvider]
+  }, [clusters, currentProvider])
+
+  const clustersLength = Object.keys(clusters).length
+  const dropdownStyle = {}
+  if (clustersLength === 1) {
+    dropdownStyle['width'] = SINGLE_PROVIDER_WIDTH
+  }
+
+  return (
+    <Dropdown
+      style={dropdownStyle}
+      button={(active, onClick) => (
+        <Dropdown.Button
+          status={
+            !!clustersLength
+              ? ComponentStatus.Default
+              : ComponentStatus.Disabled
+          }
+          active={active}
+          onClick={onClick}
+          testID="variable-type-dropdown--button"
+        >
+          {
+            regions?.find(region => region.regionId === currentRegion)
+              ?.regionName
+          }
+        </Dropdown.Button>
+      )}
+      menu={onCollapse => (
+        <Dropdown.Menu onCollapse={onCollapse}>
+          {regions?.map((c: Cluster) => (
+            <Dropdown.Item
+              key={c.regionId}
+              id={c.regionId}
+              value={c.regionId}
+              onClick={handleRegionChange}
+              testID={`variable-type-dropdown-${c.regionId}`}
+              selected={currentRegion === c.regionId}
+            >
+              {c.regionName}
```

Comment:
```suggestion
          {regions?.map((cluster: Cluster) => (
            <Dropdown.Item
              key={cluster.regionId}
              id={cluster.regionId}
              value={cluster.regionId}
              onClick={handleRegionChange}
              testID={`variable-type-dropdown-${cluster.regionId}`}
              selected={currentRegion === cluster.regionId}
            >
              {cluster.regionName}
```

I'd prefer this use variable names instead of letters, so if we ever need to search for cluster usage, we can search for `cluster` instead of `c\.`

Reply:
👍